### PR TITLE
[nrf fromlist] tests: drivers: adc: add adc_error_cases tests.

### PR DESCRIPTION
**Pull request by nordic-babu**
Tests are checking error codes returned from adc_read() and adc_setup_channel() used with invalid configurations.

Upstream PR: https://github.com/zephyrproject-rtos/zephyr/pull/76191

### CODE REVIEW

File: tests/drivers/adc/adc_error_cases/src/main.c [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+ZTEST(adc_error_cases, test_adc_read_invalid_resolution)
+{
+	int ret;
+
+	adc_channel_setup(dev_adc, &valid_channel_cfg);
+
+	struct adc_sequence invalid_seq = valid_seq;
+	/* Set resolution to invalid value */
+	invalid_seq.resolution = 99;
```
Comment (nordic-segl) [inline tests/drivers/adc/adc_error_cases/src/main.c:76]: Assuming that valid resolutions are 8-12 bits. I wonder if edge values like 7 or 13 would be better to check. However, this might be incompatible with other vendors (f.e. if their ADC supports higher/lower resolutions).

File: tests/drivers/adc/adc_error_cases/src/main.c [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+	struct adc_sequence invalid_seq = valid_seq;
+	/* Set oversampling to invalid value */
+	invalid_seq.oversampling = 99;
+
+	ret = adc_read(dev_adc, &invalid_seq);
+
+	zassert_true(
+		ret == -EINVAL,
+		"adc_read() should return True,"
```
Comment (nordic-segl) [inline tests/drivers/adc/adc_error_cases/src/main.c:56]: Why 'True'?
Return values
    0	On success.

File: tests/drivers/adc/adc_error_cases/src/main.c (hunk 1 above)
Comment (nordic-babu) [reply tests/drivers/adc/adc_error_cases/src/main.c:76]: yes, idea was to create generic TCs, as there is no restriction for resolution values in zephyr documentation 7 or 13 might be implemented by other vendors

File: tests/drivers/adc/adc_error_cases/src/main.c (hunk 2 above)
Comment (nordic-babu) [reply tests/drivers/adc/adc_error_cases/src/main.c]: good catch, it was copy-paste error, fixed.

**Review by nordic-segl — APPROVED** [review]

**Review by nordic-piks — APPROVED** [review]
